gateway-provider-ha/src/main/java/org/apache/knox/gateway/ha/dispatch/ConfigurableHADispatch.java

```diff
@@ -89,6 +91,13 @@ public void init() {
       if(stickySessionsEnabled) {
         stickySessionCookieName = serviceConfig.getStickySessionCookieName();
       }
+      /* if a configurable user agent list is defined to turn off Knox LB used it */
+      if(!StringUtils.isBlank(serviceConfig.getDisableStickySessionForUserAgents())) {
+        /* configured user agents are comma separate list which *can* contain whitespaces */
+        disableLoadBalancingForUserAgents = Arrays.asList(serviceConfig.getDisableStickySessionForUserAgents()
```

Comment:
IMO, `HaServiceConfig.getDisableStickySessionForUserAgents()` 's return type should be `List<String>` and this logic should be implemented there.

Reply:
Right, Phil mentioned the same, I'll change it.